packages/web-components/src/components/filter-panel/filter-group-item.ts

```diff
@@ -30,6 +38,190 @@ class DDSFilterGroupItem extends StableSelectorMixin(BXAccordionItem) {
+
+  /**
+   * The number of filters that can be shown without needing to hide any.
+   */
+  @property({ type: Number, attribute: 'max-filters' })
+  maxFilters: number = 7;
+
+  /**
+   * The number of filters to show when not all filters are visible.
+   */
+  @property({ type: Number, attribute: 'filter-cutoff' })
+  filterCutoff: number = 5;
+
+  /**
+   * Whether or not any hidden filters have been revealed.
+   */
+  @property({ type: Boolean })
+  allRevealed = false;
+
+  /**
+   * An element to set focus to on reveal.
+   */
+  @state()
+  _focusedElement: HTMLElement | null = null;
+
+  /**
+   * Whether or not to add view all button functionality.
+   */
+  protected _needsViewAll(): boolean {
+    return this.children.length > this.maxFilters;
+  }
+
+  /**
+   * Checks if any filters beyond the cutoff point have been selected.
+   */
+  protected _hasHiddenActiveFilter(): boolean {
+    const { children, filterCutoff } = this;
+    let result: boolean = false;
+
+    [...children].slice(filterCutoff, children.length).forEach(elem => {
+      if (elem instanceof DDSFilterPanelCheckbox) {
+        if (elem.checked) result = true;
+      }
+      if (elem instanceof DDSFilterPanelInputSelectItem || elem instanceof DDSFilterPanelInputSelect) {
+        if (elem.selected) result = true;
+      }
+    });
+
+    return result;
+  }
+
+  /**
+   * Hides or reveals excess filters.
+   */
+  protected _handleAllRevealed(revealed: boolean): void {
+    const { children, filterCutoff, accordionContent } = this;
+    const hasHiddenActiveFilter = this._hasHiddenActiveFilter();
+
+    [...children].slice(filterCutoff, children.length).forEach(elem => {
+      (elem as HTMLElement).style.display = revealed || hasHiddenActiveFilter ? '' : 'none';
+    });
+
+    if (!revealed && !hasHiddenActiveFilter) {
+      accordionContent.appendChild(this._renderViewAll());
+    }
+
+    this._dispatchViewAllEvent(revealed);
+  }
+
+  /**
+   * Generates a view all button.
+   */
+  protected _renderViewAll(): HTMLButtonElement {
+    const { children, filterCutoff } = this;
+
+    const viewAll = document.createElement('button');
+    viewAll.classList.add(viewAllClassName, `${prefix}--btn--ghost`);
+    viewAll.type = 'button';
+    viewAll.innerText = this.viewAllText;
+
+    viewAll.addEventListener(
+      'click',
+      (e): void => {
+        this.allRevealed = true;
+        if (e.target instanceof HTMLElement) e.target.remove();
+
+        const firstHidden = children[filterCutoff];
+        if (firstHidden instanceof HTMLElement) {
+          this._focusedElement = firstHidden;
+        }
+      },
+      { passive: true, once: true }
+    );
+
+    return viewAll;
+  }
+
+  /**
+   * Dispatches a custom event that notifies listeners whether or not this
+   * filter group has all options revealed.
+   */
+  protected _dispatchViewAllEvent(removed: boolean): void {
+    this.dispatchEvent(
+      new CustomEvent('filterGroupViewAll', {
```

Comment:
Can we change this event to follow the same event naming convention as other events? A name such as `filter-group-view-all-toggle` would be good. You'd also need to declare a selector in both this file and composite:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/filter-panel/filter-panel-checkbox.ts#L49